core/src/main/scala/org/apache/spark/util/random/RandomSampler.scala

```diff
@@ -41,6 +41,12 @@ trait RandomSampler[T, U] extends Pseudorandom with Cloneable with Serializable
   /** take a random sample */
   def sample(items: Iterator[T]): Iterator[U]
```

Comment:
I think if we want to keep it (which could make sense) - we should maybe add a default implementation for it so we don't have duplicate sampling logic (as we do right now in the classes)